astropy/cosmology/io/ecsv.py

```diff
@@ -9,8 +9,10 @@
 from .table import from_table, to_table
 
 
-def read_ecsv(filename, index=None, *, move_to_meta=False, cosmology=None, **kwargs):
-    """Read a `~astropy.cosmology.Cosmology` from an ECSV file.
+def read_ecsv(
+    filename, index=None, *, move_to_meta=False, cosmology=None, rename=None, **kwargs
```

Comment:
I didn't spot a single line of code that would modify `rename`, so the following should be safe and would allow some of the code to be slightly simplified:
```suggestion
    filename, index=None, *, move_to_meta=False, cosmology=None, rename={}, **kwargs
```

Reply:
I would like for this to be a safer Python feature, and I agree it is in this case, but I never use a mutable value as a default value. IMO it's too much an anti-pattern.

Reply:
If `rename` were only used in a single function then I might insist on using `{}` as the default value, but it gets passed around between quite a few functions, so using `None` as the default value is not a bad idea.